catalog/security/catalog-security-uploader-policyextension/src/main/java/ddf/catalog/security/policy/UploaderPolicyExtension.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.security.policy;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import ddf.catalog.data.Metacard;
+import ddf.security.SubjectUtils;
+import ddf.security.permission.CollectionPermission;
+import ddf.security.permission.KeyValueCollectionPermission;
+import ddf.security.permission.KeyValuePermission;
+import ddf.security.policy.extension.PolicyExtension;
+
+/**
+ * <p>
+ * Policy extension that checks for the presence of point of contact information
+ * (set by UploaderPolicyPlugin on PostQuery). If there is a point of contact
+ * and it matches the user's email, this will imply all permissions. If there is
+ * a point of contact and it does NOT match the user's email, this will only imply
+ * the point of contact permission.
+ * </p>
+ */
+public class UploaderPolicyExtension implements PolicyExtension {
+    public static final String READ_ACTION = "read";
```

Comment:
Should you use these constants here instead? https://github.com/codice/ddf/blob/master/platform/security/platform-security-core-api/src/main/java/ddf/security/permission/CollectionPermission.java#L32-L34